### ISSUE TITLE
Issue with Deploy/Release

### ISSUE THREAD

**Issue by atonse** [bonfire-networks/iconify_ex#16]
Hello,

I recently installed this library but am getting the following error only when deploying to our QA server (doing a mix release). Somehow, this worked the first time but is not working anymore.

The error: 
```elixir
** (RuntimeError) No icon set found at `/__w/foundation/foundation/deps/iconify_ex/assets/node_modules/@iconify/json/json/heroicons-solid.json` for the icon `question-mark-circle`. Find icon sets at https://icones.js.org
```

Do you know what might be causing this? If it helps, it's an umbrella app. However, it worked the first time I deployed, but hasn't worked since. And it crashes the entire app.

**Comment by mayel** [comment]
It shouldn't be looking up icons in prod, which means you need to:
- Put this in config: `config :iconify_ex, :env, config_env()` so the library knows when to prepare icons (as added to readme in a recent version).
- Have all the icons fetched/generated when in dev and make sure the output directory is included in your git commits and in the release.

Please let me know if that works for you.

**Comment by atonse** [comment]
@mayel thank you for your quick response. I've had to use inline mode to make this work (so that the generated elixir files are in code. But i don't like having all these extra modules.

When you say " make sure the output directory is included in your git commits and in the release." – what do you mean? where is the output directory if I used the preferred CSS mode? Would it be in priv? So the outputs in `priv` would have to be in source control?

I'm also encountering another issue, which is an "Unhandled throw" when I'm getting a fallback icon, even though Iconify has a catch clause for that.

<img width="538" alt="image" src="https://github.com/user-attachments/assets/980456d3-0a18-449f-b8d9-2fd907f92cef">

**Comment by mayel** [comment]
You can configure what directory to use, see https://github.com/bonfire-networks/iconify_ex/blob/main/lib%2Ficonify.ex#L168-L182 (personally I put it under /assets)

Any more details of how that throw error is appearing so I can debug?

**Comment by eriknaslund** [comment]
A backwards-incompatible change related to releases was introduced in 0.6 (https://github.com/bonfire-networks/iconify_ex/issues/15).

Before that you didn't need to specify `env: config_env()` as part of the config.

So perhaps you were using a 0.5.x release and you upgraded to a 0.6.x release?

**Comment by atonse** [comment]
Sorry I haven't responded yet. Will try a couple of your pieces of advice today or tomorrow and will report back on the progress here. It does take a while since each time I have to make an update and wait nearly 10 mins to see if it works in our QA system.

@eriknaslund we just adopted iconify_ex this week (and were delighted to see it existed) because we've been using it on a react project. so I'm not sure we've ever used 0.5.0.

**Comment by eriknaslund** [comment]
@atonse Gotcha, no rush to reply - that's the beauty of async communication :).

If you're still having issues, here's my config I'm using for reference:

```elixir
config :iconify_ex,
  env: config_env(),
  mode: :css,
  generated_icon_static_path: "./assets/css",
  default_class: nil
```

This causes a file called `assets/css/icons.css` to be generated for me.
Inside my `assets/css/app.css` I then simply `@import "icons";`, to import the generated icon CSS.

Because of this, `iconify` has everything it needs when running in `:prod` (it's all in the CSS file).

**Comment by mhelmersen-maskon** [comment]
Hi, currently looking into using the prod environment myself with iconify_ex. And from what i understand, you MUST have all the icons already generated in the icons.css file to be able to compile them into the packaged .exe file ?

Is there any way without haivng to loop through a list with prepare_icon() to generate each and every one ?  

But it seems kinda tedious to have to generate them all throught dev first, by running dev, navigating to each page so that all icons load, proceeding then to make a release for prod.

Is there something i'm missing, or am i doing something wrong ?

**Comment by mayel** [comment]
Yeah, it would affect performance if we ran the icon preparation logic on page load in prod, so unless you're using Surface for which we have a macro component that generates all needed icons at compile time, I don't know how do achieve something similar with regular LiveView... If you're using TailwindCSS though, I've seen since that iconify has an integration for it that could maybe help you instead.
